pkg/operator/controller/extension/admission/admission.go

```diff
@@ -197,11 +204,20 @@ func (d *deployment) createOrUpdateAdmissionVirtualClusterResources(ctx context.
 		}
 	}
 
-	renderedChart, err := virtualClusterClientSet.ChartRenderer().RenderArchive(archive, extension.Name, v1beta1constants.GardenNamespace, utils.MergeMaps(helmValues, gardenerValues))
+	namespace := virtualNamespace(extension)
+	renderedChart, err := virtualClusterClientSet.ChartRenderer().RenderArchive(archive, extension.Name, namespace.Name, utils.MergeMaps(helmValues, gardenerValues))
 	if err != nil {
 		return fmt.Errorf("failed rendering Helm chart %q: %w", extension.Spec.Deployment.AdmissionDeployment.VirtualCluster.Helm.OCIRepository.GetURL(), err)
 	}
 
+	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, virtualClusterClientSet.Client(), namespace, func() error {
+		metav1.SetMetaDataLabel(&namespace.ObjectMeta, v1beta1constants.GardenRole, v1beta1constants.GardenRoleExtension)
+		metav1.SetMetaDataLabel(&namespace.ObjectMeta, "extensions.operator.gardener.cloud/name", extension.Name)
+		return nil
+	}); err != nil {
+		return fmt.Errorf("failed creating namespace %q in virtual cluster: %w", namespace.Name, err)
+	}
+
 	managedResourceName := virtualManagedResourceName(extension)
```

Comment:
Is it possible to add the namespace to this `ManagedResource` instead of creating/deleting it directly in this controller?

Reply:
I don't see a nice way. We would need a [Registry](https://github.com/gardener/gardener/blob/468da8cd99c60b29a7bcf49b601e40c4157c420a/pkg/utils/managedresources/registry.go#L33) to add the namespace directly which we don't have.

The [renderedChart](https://github.com/gardener/gardener/blob/1a9e788b36578fe213b74309b71db53c0d81dad0/pkg/operator/controller/extension/admission/admission.go#L208) does not provide a way to add another resource to it.
If we serialize the namespace manually and add it to the secret of the MR, the whole thing would be more complicated than it currently is.

Reply:
Hm, what's so bad about creating the registry for it? We do it similarly for the Istio deployment [here](https://github.com/gardener/gardener/blob/ed4c82e3732ab2160686beba44d1bcb51767f31a/pkg/component/networking/istio/istiod.go#L145).
This might be a little overhead but therefore signatures and the deletion logic get simpler.

Reply:
Ah, I did not know that 🙂

Reply:
It doesn't seem to work. I'll check that later.
Btw. in the istio deployment case you referenced, the namespace is created separately.

Reply:
The MR of the virtual part of admission was stuck in deletion. I adapted the `Garden` deletion flow to unstuck it.
Details can be found the in the commit message (https://github.com/gardener/gardener/pull/10585/commits/e4cc39872f2187893dcf0b8f2696f5e4db17ee5d).

Reply:
I reverted the change of moving `gardener-apiserver` because this created a new dependency to the availability of `gardener-controller-manager` (https://github.com/gardener/gardener/pull/10585/commits/f33c17c5d4a04339984d04388b1a106d32a6a9b3).
`IgnoreNotFound` in GNA seems to cover the case, that the entire Kind is gone.